File: src/main/java/backend/Person.java
package backend;

public class Person {
    private String firstName;
    private String lastName;
    private int age;
    private long pesel;
    Person(){

    };
    public Person(String firstName, String lastName)throws NullNameException {
        if(firstName.length() > 0 && lastName.length() > 0){
            this.firstName = firstName;
            this.lastName= lastName;
        }else throw new NullNameException("Imię i nazwisko nie mogą być puste");
    }
    public Person(String firstName, String lastName, long pesel) throws PeselException, NullNameException {
        this.firstName = firstName;
        this.lastName = lastName;
        if(firstName.length() > 0 && lastName.length() > 0){
            this.firstName = firstName;
            this.lastName = lastName;}else throw new NullNameException("Imię i nazwisko nie mogą być puste");
        if(pesel > 9999999999L){this.pesel = pesel;}else throw new PeselException("Pesel musi mieć 11 cyfr");
    }

    public Person(String firstName, String lastName, long pesel, int age)throws PeselException, NullNameException {
        if(firstName.length() > 0 && lastName.length() > 0) {this.firstName = firstName;
        this.lastName = lastName;}else throw new NullNameException("Imię i nazwisko nie mogą być puste");
        if(pesel > 9999999999L){this.pesel = pesel;}else throw new PeselException("Pesel musi mieć 11 cyfr");
        this.age = age;
    }

    public String getFirstName() {
        return firstName;
    }

    public int getAge() {
        return age;
    }

    public String getLastName() {
        return lastName;
    }

    public long getPesel() {
        return pesel;
    }

    public void setFirstName(String firstName) throws NullNameException {
        if(firstName.length()>0)this.firstName = firstName;
        else throw new NullNameException("Imię nie może być puste");
    }

    public void setLastName(String lastName) throws NullNameException {
        if(lastName.length()>0)this.lastName = lastName;
        else throw new NullNameException("Nazwisko nie może być puste");
    }

    public void setAge(int age) {
        this.age = age;
    }

    public void setPesel(long pesel)throws PeselException {
        if(pesel > 9999999999L){this.pesel = pesel;}else throw new PeselException("Pesel musi mieć 11 cyfr");

    }
}

File: JS_DLA_MONGO/klasy/person.js
db.system.js.save({
    _id: "Person",  // Nazwa funkcji
    value: function(firstName, lastName, pesel, age) {
        // Walidacja
        if (!firstName || firstName.trim().length === 0 || !lastName || lastName.trim().length === 0) {
            throw new Error("Imię i nazwisko nie mogą być puste");
        }

        if (pesel.toString().length !== 11 || pesel < 0) {
            throw new Error("Pesel musi składać się dokładnie z 11 cyfr");
        }

        if (age < 0) {
            throw new Error("Wiek nie może być ujemny");
        }

        // Zwracamy obiekt, który emuluje instancję klasy Person
        return {
            firstName: firstName,
            lastName: lastName,
            pesel: pesel,
            age: age,

            getFirstName: function() {
                return this.firstName;
            },
            getLastName: function() {
                return this.lastName;
            },
            getPesel: function() {
                return this.pesel;
            },
            getAge: function() {
                return this.age;
            },
            setFirstName: function(newFirstName) {
                if (!newFirstName || newFirstName.trim().length === 0) {
                    throw new Error("Imię nie może być puste");
                }
                this.firstName = newFirstName;
            },
            setLastName: function(newLastName) {
                if (!newLastName || newLastName.trim().length === 0) {
                    throw new Error("Nazwisko nie może być puste");
                }
                this.lastName = newLastName;
            },
            setPesel: function(newPesel) {
                if (newPesel.toString().length !== 11 || newPesel < 0) {
                    throw new Error("Pesel musi składać się dokładnie z 11 cyfr");
                }
                this.pesel = newPesel;
            },
            setAge: function(newAge) {
                if (newAge < 0) {
                    throw new Error("Wiek nie może być ujemny");
                }
                this.age = newAge;
            }
        };
    }
});
